modules/community-algorithms/pages/k-means-clustering.adoc

```diff
@@ -0,0 +1,83 @@
+= k-Means Clustering
+
+.Supported Graph Characteristics
+****
+[cols='1']
+|===
+^|Unweighted edges
+^|Weighted edges
+^|Directed edges
+^|Undirected edges
+^|Homogeneous vertex types
+|===
+
+Algorithm link: link:https://github.com/tigergraph/gsql-graph-algorithms/tree/master/algorithms/Community/k_means[k-Means Clustering]
+
+****
+
+In the k-Means Clustering algorithm, vertices are gradually sorted into an arbitrary number _k_ of clusters based on averaging their properties.
+
+First, _k_ vertices are selected at random to act as centroids, or the center of their clusters. The first clusters are defined as consisting of the vertices closest to those centroids. The borders of these clusters are refined iteratively by calculating the mean position of all points in the clusters, then by finding the closest vertices to those mean position points to act as the new centroids.
+
+The last two steps are repeated many more times until each iteration no longer changes the location of the centroids or the vertices belonging to the clusters.
+
+Vertices in the graph need to have an attribute in their schemas that is a list of doubles or floats (an embedding) for this algorithm to work. This is because the distances are calculated based on the vertex embeddings rather than any other factor.
+
+In some use cases, an embeddings algorithm like FastRP can be used to create these embeddings.
+If you have vertices with separate latitude and longitude attributes, you must first prepare the data by combining the two coordinates into a list of `FLOAT` values in a single attribute.
+
+== Notes
+
+This algorithm ignores edges in the graph and looks only at vertex attributes.
+
+This query uses another subquery https://github.com/tigergraph/gsql-graph-algorithms/blob/master/algorithms/Community/k_means/tg_kmeans_sub.gsql[`tg_kmeans_sub`], which needs to be installed before `tg_kmeans` can be installed.
```

Comment:
We can mention that you can use the one-click install to do this in graphstudio, and i don't think you need to install subqueries if you do it that way? We should also update the https://docs.tigergraph.com/graph-ml/current/using-an-algorithm/#_install_query page to mention this